scheduler/src/cook/compute_cluster.clj

```diff
@@ -46,21 +46,20 @@
              [?c :compute-cluster/mesos-framework-id ?framework-id?]]
            unfiltered-db compute-cluster-name framework-id)
         materialized-query-result (into [] query-result)]
-    materialized-query-result))
+    (first materialized-query-result)))
 
 (defn get-mesos-cluster-map
   "Process one mesos cluster specification, returning the entity id of the corresponding compute-cluster, creating the cluster if it does not exist."
   [conn {:keys [compute-cluster-name framework-id] :as mesos-cluster}]
   {:pre [compute-cluster-name
          framework-id]}
   (let [cluster-entity (get-mesos-cluster-entity (d/db conn) mesos-cluster)]
-    (when (zero? (count cluster-entity))
+    (when-not cluster-entity
       (create-missing-compute-cluster conn (mesos-cluster->compute-cluster-map-for-datomic mesos-cluster)))
     {:cluster-type :mesos-cluster
      :compute-cluster-name compute-cluster-name
      :mesos-framework-id framework-id
-     :db-id (->> (get-mesos-cluster-entity (d/db conn) mesos-cluster)
-                 first)}))
+     :db-id (get-mesos-cluster-entity (d/db conn) mesos-cluster)}))
```

Comment:
Can the call to `get-mesos-cluster-entity` be replaced with `cluster-entity`?

Reply:
Renamed.

Reply:
No. It it didn't exist the first time around, we need to requery it to make sure we got the one we JUST inserted.